src/tests/ftest/cart/utest/utest_portnumber.c

```diff
@@ -0,0 +1,286 @@
+ * limitations under the License.
+ *
+ * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+ * The Government's rights to use, modify, reproduce, release, perform, display,
+ * or disclose this software are subject to the terms of the Apache License as
+ * provided in Contract No. 8F-30005.
+ * Any reproduction of computer software, computer software documentation, or
+ * portions thereof marked with this legend must also reproduce the markings.
+ */
+/**
+ * This file is part of CaRT testing.
+ */
+
+/*
+ * This code tests whether a provider returns an error if 2 independent
+ * instances attempts to open the same port number.
+ *
+ * Two child process are forked with the same provider information.
+ * The test is setup so that first child opens the port and then
+ * the second child should fail.
+ * Note, synchronization between child process is performed via sleep
+ * function calls.  The sleep time should prevent any problems, but
+ * beware if an issue arise where the results are swapped.
+ */
+/*
+ * Reference jira ticket DAOS-5732 to include socket and verb tests.
+ */
+#define MY_TESTS_NOT_INCLUDED
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <stdarg.h>
+#include <stddef.h>
+#include <setjmp.h>
+#include <time.h>
+#include <sys/wait.h>
+
+#include <sys/types.h>
+#include <sys/shm.h>
+#include <sys/ipc.h>
+#include <semaphore.h>
+#include <string.h>
+#include <errno.h>
+
+#include <cmocka.h>
+#include <cart/api.h>
+#include "gurt/debug.h"
+
+/* global semaphores */
+sem_t *child1_sem;
+sem_t *child2_sem;
+
+int shmid_c1;
+int shmid_c2;
+
+static void
+run_test_fork(void **state)
+{
+	int		result1 = -1;
+	int		result2 = -1;
+	int		status = -1;
+	int		rc = 0;
+	int		child_result;
+	pid_t		pid1 = 0;
+	pid_t		pid2 = 0;
+	crt_context_t	crt_context = NULL;
+
+	/* lock the semaphore */
+	sem_trywait(child1_sem);
+	sem_trywait(child2_sem);
+
+	/* fork first child process */
+	pid1 = fork();
+	if (pid1 == 0) {
+		rc = crt_init(NULL, CRT_FLAG_BIT_SERVER);
+		if (rc != 0) {
+			sem_post(child2_sem);
+			rc = 10;
+			goto child1_error;
+		}
+		child_result = crt_context_create(&crt_context);
+
+		/* Signal second child to continue and wait */
+		sem_post(child2_sem);
+		sem_wait(child1_sem);
+
+		if (child_result == 0) {
+			rc = crt_context_destroy(crt_context, false);
+			if (rc != 0) {
+				rc = 11;
+				goto child1_error;
+			}
+		}
+		rc = crt_finalize();
+		if (rc != 0) {
+			rc = 12;
+			goto child1_error;
+		}
+		exit(child_result);
+child1_error:
+		exit(rc);
+	}
+
+	/* fork second child process */
+	pid2 = fork();
+	if (pid2 == 0) {
+		/* wait for signal from child 1 */
+		sem_wait(child2_sem);
+
+		rc = crt_init(NULL, CRT_FLAG_BIT_SERVER);
+		if (rc != 0) {
+			rc = 20;
+			goto child2_error;
+		}
+		child_result = crt_context_create(&crt_context);
+		if (child_result == 0) {
+			rc = crt_context_destroy(crt_context, false);
+			if (rc != 0) {
+				rc = 21;
```

Comment:
should call crt_finalize() here before jumping to exit label

Reply:
Fixed.